### PR TITLE
Start a CONSUMER span for Kafka poll(); and refactor spring-kafka...

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
... so that it has 1 batch receive span + 1 batch process span. This basically implements this idea: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/3922#discussion_r696407770 (well, a part of it).

The process CONSUMER span is not a child of the receive CONSUMER span yet (spec [says](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#batch-receiving) it should be) because successive CONSUMERs are suppressed; but I'm planning to introduce a new suppression strategy for consumer spans in the next PR (separate span keys for receive an process).

This PR is split into 2 commits: the first contains just `kafka-clients` changes, the second just `spring-kafka` changes. Maybe that'll help reviewing it.

### CODE REVIEW

File: instrumentation/kafka-clients/kafka-clients-0.11/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kafkaclients/ReceivedRecords.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  public abstract ConsumerRecords<?, ?> records();
+
+  abstract Timer timer();
+
+  public Instant startTime() {
+    return timer().startTime();
+  }
+
+  public Instant endTime() {
+    return timer().endTime();
```
Comment (trask) [inline instrumentation/kafka-clients/kafka-clients-0.11/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kafkaclients/ReceivedRecords.java:28]: maybe
```suggestion
  public Instant now() {
    return timer().now();
```

**Review by trask — APPROVED** [review]
I think it would be nice if we can avoid releasing in the intermediate state where we are creating disjointed consumer spans

**Comment by mateuszrzeszutek** [comment]
> I think it would be nice if we can avoid releasing in the intermediate state where we are creating disjointed consumer spans

Yeah, definitely - I'll start working on the next part soon, we should definitely get it merged before the next release.

**Comment by arm-chair** [comment]
Thanks @mateuszrzeszutek, were you able to make any progress on the next part? I think I am maybe seeing the disjointed consumer in an downstream Lightstep alpha release.

**Comment by trask** [comment]
@arm-chair does lightstep backend support links? if so you should be seeing the consumer linked to the producer.
if not, check out #4187

**Comment by mateuszrzeszutek** [comment]
Span links were added here: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/4122 and https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/4151

**Pull request merged**